archeryutils/handicaps/handicap_equations.py

```diff
@@ -355,15 +356,126 @@ def sigma_r(
     return sig_r
 
 
+def s_bar(
+        target_specs: dict[float, int],
+        arw_rad: float,
+        sig_r: FloatArray
+    ) -> FloatArray:
+    """Calculate expected score directly from target ring sizes.
+
+    Parameters
+    ----------
+    target_specs : dict[float, int]
+        Mapping of target ring *diameters* in [metres], to points scored
+    arw_d : float
+        arrow diameter in [metres]
+    sig_r : float
+        standard deviation of group size [metres]
+
+    Returns
+    -------
+    s_bar : float
+        expected average score per arrow
+
+    Notes
+    -----
+    May differ from previous implementions due to floating point errors in deriving
+    ring radiuses directly from diameters, rather than multiplying by ring count and
+    dividing by overall diameter
+    Eg 0.12 / 2 != 3*0.4 / 20
+
+    Examples
+    --------
+    >>> #WA 18m compound triple spot
+    >>> specs = {0.02: 10, 0.08: 9, 0.12: 8, 0.16: 7, 0.2: 6}
+    >>> _sbar(specs, 9.3e-3, 0.04)
+    8.928787288284521 # differs from previous implementation by 1.7763568394002505e-15
+    """
+
+
+    ring_scores = sorted(itr.chain(target_specs.values(), [0]), reverse=True)
+    score_drops = [inner - outer for inner,outer in itr.pairwise(ring_scores)]
+    max_score = max(ring_scores)
+
+    return max_score - sum(
+        score_drop * np.exp(-(((arw_rad + (ring_diam / 2)) / sig_r) ** 2))
+        for ring_diam, score_drop in zip(target_specs, score_drops)
+    )
+
+
+def get_spec_from_target(target: targets.Target) -> dict[float,int]:
+    """Derive specifications for common/supported targets
+
+    Parameters
+    ----------
+    target : Target
+        Target object that has a defined scoring system and diameter
+
+    Returns
+    -------
+    spec : dict
+        Mapping of target ring sizes in [metres] to score
+    """
+    # 1 too many branches. Makes sense due to different target faces => disable
+    # Could replace with mapping face to lambda that returns spec
+    # pylint: disable=too-many-branches
+    def _rnd6(value):
+        return round(value, 6)
```

Comment:
Is there a reason to round to 6 dp?

Reply:
6 was aribtrary, I think from dividing the worcster face which is already defined to 4 s.f. 
But its to avoid floating point shenanigans with the construction of the ring sizes, which give different results depending how you get there.

Eg:
```python
# five_zone target
# divide diameter by 10, multiply by ring number
# eg red is 4 * 0.8 / 10
>>> tar_dia = 0.8
>>> {(n+1)*tar_dia / 10 : 10-n for n in range(1,11,2)}
{0.16: 9, 0.32: 7, 0.4800000000000001: 5, 0.64: 3, 0.8: 1}

# multiply by fraction of target size 
# eg red is 0.4 * 0.8
>>> {((n+1) / 10) * tar_dia : 10-n for n in range(1,11,2)}
{0.16000000000000003: 9,
 0.32000000000000006: 7,
 0.48: 5,
 0.6400000000000001: 3,
 0.8: 1}

# with rounding doesn't matter which order multiplication/division done in and all diameters come out sensibly
>>> {round((n+1)*tar_dia / 10, 6) : 10-n for n in range(1,11,2)}
{0.16: 9, 0.32: 7, 0.48: 5, 0.64: 3, 0.8: 1}
```

Reply:
Aaahhhh, I misunderstood this before - I thought it was affecting the returned score, but I realise its in setting up the target diameters.

A quick experiments suggests its on machine epsilon (10^-16) as rounding to 15 dp give the expected value.

6 doesn't seem unreasonable given tests still pass provided it is explained/documented somewhere.

Reply:
Yeah, I think given the target faces themselves are made to [tolerances](https://www.worldarchery.sport/rulebook/article/13) of ~1mm on the inner rings, rounding to the nearest micron should be more than enough accuracy. Agree need to document.